lib/src/signed_video_h26x_common.c

```diff
@@ -928,14 +928,18 @@ update_hash(signed_video_t *self,
 }
 
 static svrc_t
-set_linked_hash(signed_video_t *self, uint8_t *hash, size_t hash_size)
+update_linked_hash(signed_video_t *self, uint8_t *hash, size_t hash_size)
 {
   if (!self || !hash) return SV_INVALID_PARAMETER;
+
   gop_info_t *gop_info = self->gop_info;
-  memcpy(gop_info->linked_hash_data.linked_hash, gop_info->linked_hash_data.stored_hash,
-      self->sign_data->hash_size);
-  uint8_t *linked_hash = gop_info->linked_hash_data.stored_hash;
-  memcpy(linked_hash, hash, hash_size);
+  uint8_t *stored_hash = &gop_info->linked_hashes[hash_size];
+  uint8_t *linked_hash = &gop_info->linked_hashes[0];
+  // Move the content of stored_hash to linked_hash
+  memmove(linked_hash, stored_hash, self->sign_data->hash_size);
+  // Copy the new hash to stored_hash
+  memcpy(stored_hash, hash, self->sign_data->hash_size);
```

Comment:
Here you must use `hash_size` in case it differs with `sign_data->hash_size`.
It should not differ, so you could/should probably add a sanity comparison check for it.